test/data/electrical.json

```diff
@@ -154,14 +157,133 @@
                 "$source": "foo.bar",
                 "value": 230
               }
+            },
+            "inverterMode": {
+              "timestamp": "2014-08-15T19:00:15.402Z",
+              "$source": "foo.bar",
+              "value": "idle"
+            }
+          }
+        },
+        "alternator": {
+          "main": {
+            "location": "Mains - (house) alternator",
+            "manufacturer": {
+              "name": "Leece Neville + Arduino Alt Reg",
+              "model": "4870 + 0.3.6",
+              "URL":  "https://arduinoalternatorregulator.blogspot.com/"         
+            },
+            "associatedBus": "house1",
+            "chargingAlgorithm": {
+              "timestamp": "2014-08-15T19:00:15.402Z",
+              "$source": "foo.bar",
+              "value": "three stage"
+            },
+            "chargerMode": {
+              "timestamp": "2014-08-15T19:00:15.402Z",
+              "$source": "foo.bar",
+              "value":  "slave"
+            },
+            "mode": {
+              "timestamp": "2014-08-15T19:00:15.402Z",
+              "$source": "foo.bar",
+              "value": "charging bulk"
+            },
+            "voltage": {
+              "timestamp": "2014-08-15T19:00:15.402Z",
+              "$source": "foo.bar",
+              "value": 14.13
+            },
+            "setpointVoltage": {
+              "timestamp": "2014-08-15T19:00:15.402Z",
+              "$source": "foo.bar",
+              "value": 14.52
+            },            
+            "current": {
+              "timestamp": "2014-08-15T19:00:15.402Z",
+              "$source": "foo.bar",
+              "value": 210.5
+            }
+          }
+        },
+        "solar": {
+          "port240": {
+            "location": "Port array",
+            "manufacturer": {
+              "name": "REC + Morningstar",
+              "model": "REC-240 + Tristar MPPT45"      
+            },
+            "associatedBus": "house1",
+            "chargingAlgorithm": {
+              "timestamp": "2014-08-15T19:00:15.402Z",
+              "$source": "foo.bar",
+              "value": "three stage"
+            },
+            "chargerMode": {
+              "timestamp": "2014-08-15T19:00:15.402Z",
+              "$source": "foo.bar",
+              "value":  "standalone"
+            },
+            "mode": {
+              "timestamp": "2014-08-15T19:00:15.402Z",
+              "$source": "foo.bar",
+              "value": "charging acceptance"
+            },
+            "voltage": {
+              "timestamp": "2014-08-15T19:00:15.402Z",
+              "$source": "foo.bar",
+              "value": 14.3
+            },            
+            "current": {
+              "timestamp": "2014-08-15T19:00:15.402Z",
+              "$source": "foo.bar",
+              "value": 12.2
+            }
+          }
+        },
+        "solar": {
```

Comment:
Port and Starboard array, showing two different controllers.  Is the form incorrect for that?